### PR TITLE
feat: payment-services endpoint should return only active or deleted [TA-2401]

### PR DESCRIPTION
**Pull request by Dave-Elec**
# Description

This PR updates /payment-services endpoint to include `deleted` query parameter.
This allows either deleted or active payment services to be listed.

Jira: [TA-2401](https://gr4vy.atlassian.net/browse/TA-2401)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own changes
- [x] I have run `yarn test` to make sure my changes pass all linters
- [x] I have pulled the latest changes from the upstream `main` branch

## Contribution guidelines

For contribution guidelines, styleguide, and other helpful information please
see the `CONTRIBUTING.md` file in the root of this project.

### CODE REVIEW

File: reference/paths/payment_services__list-payment-services.yaml [hunk 1]
```diff
@@ -9,6 +9,7 @@ parameters:
   - $ref: "../query-parameters/limit-100.yaml"
   - $ref: "../query-parameters/cursor.yaml"
   - $ref: "../query-parameters/method.yaml"
+  - $ref: "../query-parameters/list_payment_services__deleted.yaml"
```
Comment (cbetta) [inline reference/paths/payment_services__list-payment-services.yaml:12]: Can we rename this to `deleted.yml`?

File: reference/query-parameters/deleted.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+---
+schema:
+  type: boolean
+  example: false
```
Comment (thomas-gr4vy) [inline reference/query-parameters/deleted.yaml:4]:
```suggestion
  example: true
  default: false
```

File: reference/query-parameters/deleted.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+in: query
+name: deleted
+description: |-
+  When set to `true`, filter for payment services that are deleted.
+  When set to `false`, filter for payment services that are active.
```
Comment (thomas-gr4vy) [inline reference/query-parameters/deleted.yaml:9]:
```suggestion
  Filters the results to only show items which have been deleted.
  By default, deleted items will not be returned.
```

File: reference/query-parameters/deleted.yaml (hunk 2 above)
Comment (thomas-gr4vy) [reply reference/query-parameters/deleted.yaml:4]: As a user is unlikely to ever use `deleted=false` (it's unnecessary) we should instead show the intended usage of filtering to only show deleted items by settings `deleted=true`.

File: reference/query-parameters/deleted.yaml (hunk 3 above)
Comment (thomas-gr4vy) [reply reference/query-parameters/deleted.yaml:9]: We made this parameter generic when we renamed it to `deleted` so we should make the description generic also.

**Review by thomas-gr4vy — CHANGES_REQUESTED** [review]

**Review by thomas-gr4vy — APPROVED** [review]
Looks good to me, as long as @cbetta is happy too.

**Review by cbetta — APPROVED** [review]

**Pull request merged**